src/apps/settings/src/app/views/User/Workflows/authorized/forms-dialogs/StatusLabelForm.tsx

```diff
@@ -0,0 +1,519 @@
+  ) => setSelectedColor(newValue.map((item) => item.value).join(","));
+
+  return (
+    <>
+      <Autocomplete
+        multiple
+        fullWidth
+        options={sortedListData}
+        disableCloseOnSelect
+        value={getRoleInfo(value)}
+        onChange={handleChange}
+        size="small"
+        getOptionLabel={(option) => option.label}
+        renderOption={(props, option, { selected }) => (
+          <li {...props}>
+            <Checkbox
+              size="small"
+              disableRipple
+              icon={<CheckIcon fontSize="medium" sx={{ opacity: 0 }} />}
+              checkedIcon={<CheckIcon fontSize="medium" />}
+              checked={selected}
+              sx={{
+                marginRight: 5,
+                position: "absolute",
+                left: 0,
+              }}
+            />
+            <Typography variant="body2" sx={{ pl: 4 }}>
+              {option.label}
+            </Typography>
+          </li>
+        )}
+        renderInput={(params) => (
+          <TextField
+            {...params}
+            variant="outlined"
+            placeholder={value ? "" : "None"}
+          />
+        )}
+        ChipProps={{
+          sx: {
+            height: "1.25rem",
+            backgroundColor: "transparent",
+            outline: "1px solid",
+            outlineColor: (theme) => theme.palette.border,
+          },
+        }}
+      />
+      <input type="hidden" name={name} value={value?.trim()} />
+    </>
+  );
+};
+
+const validateFormData = (formData: WorkflowStatus.CreateStatusLabel) => {
+  const errors: Record<string, string> = {};
+  if (!formData.name) errors.name = "Name is required";
+  // if (!formData.description) errors.description = "Description is required";
+  if (!formData.color) errors.color = "Color is required";
+  return errors;
+};
+
+const StatusLabelForm: FC<StatusLabelFormProps> = ({
+  open,
+  onClose,
+  labels = [],
+  values,
+  isDeactivated = false,
+}) => {
+  const ZUID = values?.ZUID || undefined;
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [rolesMenuItems, setRolesMenuItems] = useState<
+    WorkflowStatus.StatusLabelsRoleMenu[]
+  >([]);
+  const {
+    isLoading: rolesLoading,
+    isFetching,
+    data: rolesMenuData,
+  } = useGetUsersRolesQuery();
+
+  const [formErrors, setFormErrors] = useState<Record<string, string>>({});
+  const dispatch = useDispatch();
+  const [createWorkflowStatusLabel] = useCreateWorkflowStatusLabelMutation();
+  const [updateWorkflowStatusLabel] = useUpdateWorkflowStatusLabelMutation();
+  const { openDeactivationDialog, setFocusedLabel } = useFormDialogContext();
+
+  const usedColors = labels.map((label) => label.color);
+
+  const transformRoleValues = (value: string): string[] =>
+    value ? value.split(",") : [];
+
+  const handleFormSubmit = async (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    setIsLoading(true);
```

Comment:
The mutation queries already have an `isLoading` so you won't have to manually handle this anymore

Reply:
The query's "isLoading" variable is out of scope. It is triggered inside a popup modal, where it controls the loading state for the asynchronous onSubmit function. This ensures that the modal does not close and displays a loading indicator on the button while the function is being executed.